services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/thumbnails/ThumbnailRequestProcessor.java

```diff
@@ -39,17 +40,26 @@ public void process(Exchange exchange) throws IOException {
                 pid, agent.getPrincipals(), Permission.editDescription);
 
         var file = repositoryObjectLoader.getFileObject(pid);
-        var work = file.getParent();
+        var work = (WorkObject) file.getParent();
 
         if (Objects.equals(action, ThumbnailRequest.ASSIGN)) {
+            // Capture the old thumbnail id before it gets cleared
+            var oldThumbnailFile = work.getThumbnailObject();
             repositoryObjectFactory.createExclusiveRelationship(work, Cdr.useAsThumbnail, file.getResource());
+            // reindex old thumbnail object
+            if (oldThumbnailFile != null) {
+                indexingMessageSender.sendIndexingOperation(
+                        agent.getUsername(), oldThumbnailFile.getPid(), IndexingActionType.UPDATE_DATASTREAMS);
+            }
         } else if ( Objects.equals(action, ThumbnailRequest.DELETE)) {
             repositoryObjectFactory.deleteProperty(work, Cdr.useAsThumbnail);
         }
 
         // send message to update solr
         indexingMessageSender.sendIndexingOperation(
                 agent.getUsername(), work.getPid(), IndexingActionType.UPDATE_DATASTREAMS);
+        indexingMessageSender.sendIndexingOperation(
```

Comment:
These changes all look good, just wondering if the test for this might need updating to verify that the additional indexing messages are sent?